### PR TITLE
CSP-735 Calendar Events Search keyword filter

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Models/GetCalendarEventsRequestModel.cs [hunk 1]
```diff
@@ -10,6 +10,9 @@ public class GetCalendarEventsRequestModel
     [FromHeader(Name = Constants.ApiHeaders.RequestedByMemberIdHeader)]
     public Guid RequestedByMemberId { get; set; }
 
+    [FromQuery]
+    public string Keyword { get; set; } = string.Empty;
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Models/GetCalendarEventsRequestModel.cs:14]: This should be nullable

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQuery.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@ namespace SFA.DAS.ApprenticeAan.Application.CalendarEvents.Queries.GetCalendarEv
 public class GetCalendarEventsQuery : IRequest<GetCalendarEventsQueryResult>
 {
     public Guid RequestedByMemberId { get; set; }
+    public string Keyword { get; set; } = string.Empty;
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQuery.cs:8]: This should be nullable

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Services/QueryStringParameterBuilder.cs [hunk 3]
```diff
@@ -8,6 +8,7 @@ public static class QueryStringParameterBuilder
     public static Dictionary<string, string[]> BuildQueryStringParameters(GetCalendarEventsQuery request)
     {
         var parameters = new Dictionary<string, string[]>();
+        if (request.Keyword.Trim() != string.Empty) parameters.Add("keyword", new[] { request.Keyword });
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Services/QueryStringParameterBuilder.cs:11]: The input should be trimmed once at the source (web) and then there should be no need to trim it down the line.

Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Services/QueryStringParameterBuilder.cs:11]: ok cool.  I'll use a string.IsNullOrWhitespace here

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQuery.cs (hunk 2 above)
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/CalendarEvents/Queries/GetCalendarEvents/GetCalendarEventsQuery.cs:8]: ok

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**